apps/etterlatte-vilkaarsvurdering/src/main/kotlin/vilkaarsvurdering/VilkaarsvurderingService.kt

```diff
@@ -76,16 +96,35 @@ class VilkaarsvurderingService(
                                 grunnlagVersjon = grunnlag.metadata.versjon
                             )
                         )
+
                     else ->
                         throw IllegalArgumentException(
                             "Støtter ikke vilkårsvurdering for behandlingType=${behandling.behandlingType}"
                         )
                 }
+
             SakType.OMSTILLINGSSTOENAD ->
                 throw IllegalArgumentException("Støtter ikke vilkårsvurdering for sakType=$sakType")
         }
     }
 
+    private suspend fun sjekkOgCommitVilkaarsvurdering(
+        behandlingId: UUID,
+        accessToken: String,
+        block: suspend () -> Vilkaarsvurdering
```

Comment:
Block er litt for generisk her kanskje, hva om den heter feks `opprettVilkaarsvurdering`?

Reply:
poenget er vel at den skal være ganske generisk, slik at man kan bruke metoden til både oppretting, sletting osv. Kan være at hele metoden bør ha et annet navn evt?

Reply:
den kunne hatt ett bedre navn ja, skal se om jag kommer på noe mer beskrivende.